src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8106,25 +8106,19 @@ private SwitchSectionSyntax ParseSwitchSection()
                         else
                         {
                             var node = CheckRecursivePatternFeature(ParseExpressionOrPattern(whenIsKeyword: true, forSwitchCase: true, precedence: Precedence.Ternary));
-                            if (this.CurrentToken.ContextualKind == SyntaxKind.WhenKeyword)
-                            {
-                                if (node is ExpressionSyntax)
-                                {
-                                    // if there is a 'when' token, we treat a case expression as a constant pattern.
-                                    node = _syntaxFactory.ConstantPattern((ExpressionSyntax)node);
-                                }
-                            }
+
+                            // if there is a 'when' token, we treat a case expression as a constant pattern.
+                            if (this.CurrentToken.ContextualKind == SyntaxKind.WhenKeyword && node is ExpressionSyntax ex)
+                                node = _syntaxFactory.ConstantPattern(ex);
 
                             if (node.Kind == SyntaxKind.DiscardPattern)
```

Comment:
I believe our style guide requires braces in all cases.

Reply:
No.  I checked.